src/aspects/ec2-automated-shutdown/handler/index.ts

```diff
@@ -0,0 +1,102 @@
+import { Context, CloudWatchAlarmData } from 'aws-lambda';
+import { EC2, StopInstancesCommandOutput } from '@aws-sdk/client-ec2';
+
+interface CloudWatchEvent {
+    version: string;
+    id: string;
+    'detail-type': string;
+    source: string;
+    account: string;
+    time: string;
+    region: string;
+    resources: string[];
+    detail: CloudWatchAlarmData;
+}
+
+interface MetricStat {
+    metric: {
+        dimensions?: Array<{
+            name: string;
+            value: string;
+        }>;
+    };
+}
+
+interface AlarmConfiguration {
+    metrics?: Array<{
+        metricStat?: MetricStat;
+    }>;
+}
+
+const getInstanceId = (event: CloudWatchEvent): string => {
+    const configuration = event.detail.configuration as AlarmConfiguration;
+    if (!configuration.metrics?.[0]?.metricStat?.metric?.dimensions) {
+        throw new Error('Instance ID not found in alarm metrics');
+    }
+
+    const dimensions = configuration.metrics[0].metricStat.metric.dimensions;
+
+    if (!!dimensions.InstanceId) {
```

Comment:
I'm getting `Property 'InstanceId' does not exist on type '{ name: string; value: string; }[]'.ts(2339)` on this property. Is this currently working?